build.mk

```diff
@@ -0,0 +1,84 @@
+OS = $(shell uname | tr A-Z a-z)
+GOARCH = $(shell go env GOARCH)
+GOLANG_VERSION ?= 1.18
+
+# Build variables
+BUILD_DIR ?= bin
+BINARY_NAME_client ?= p1
+BINARY_NAME_pocket ?= pocket
+POST_BUILD_TARGETS = rename-binaries
+VERSION ?= $(shell git describe --tags --exact-match 2>/dev/null || git symbolic-ref -q --short HEAD)
+COMMIT_HASH ?= $(shell git rev-parse --short HEAD 2>/dev/null)
+DATE_FMT = +%FT%T%z
+ifdef SOURCE_DATE_EPOCH
+    BUILD_DATE ?= $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u "$(DATE_FMT)")
+else
+    BUILD_DATE ?= $(shell date "$(DATE_FMT)")
+endif
+LDFLAGS += -X main.version=${VERSION} -X main.commitHash=${COMMIT_HASH} -X main.buildDate=${BUILD_DATE}
+export CGO_ENABLED ?= 0
+ifeq (${VERBOSE}, 1)
+ifeq ($(filter -v,${GOARGS}),)
+	GOARGS += -v
+endif
+endif
+
+.PHONY: clean
+clean: ${CLEAN_TARGETS} ## Clean the build directory
+	rm -rf ${BUILD_DIR}/
+
+.PHONY: goversion
+goversion: ## Check if the installed Go version is the required one
+ifneq (${IGNORE_GOLANG_VERSION}, 1)
+	@printf "${GOLANG_VERSION}\n$$(go version | awk '{sub(/^go/, "", $$3);print $$3}')" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -g | head -1 | grep -q -E "^${GOLANG_VERSION}$$" || (printf "Required Go version is ${GOLANG_VERSION}\nInstalled: `go version`" && exit 1)
+endif
+
+.PHONY: build-%
+build-%: pre-build
+build-%: goversion ## Build a single application in the app directory
+ifeq (${VERBOSE}, 1)
+	go env
```

Comment:
Why print `go env` when verbose?

Reply:
Just so the user can see the golang environment being used to build the binary. I thought that might be good to add since some developers might be using multiple golang versions at once with something like https://github.com/syndbg/goenv

Reply:
```suggestion
	go env ## Informs developer of go environment
```

Reply:
https://github.com/pokt-network/pocket/pull/390/files#r1050251471